generated/serialization.h

```diff
@@ -0,0 +1,39 @@
+#ifndef DTOGEN_SERIALIZATION_H
+#define DTOGEN_SERIALIZATION_H
+
+#include "nlohmann/json.hpp"
+
+#include <optional>
+#include <memory>
+
+namespace nlohmann {
+
+template <typename T>
+struct adl_serializer<std::optional<T>> {
+    static void to_json(json& j, const std::optional<T>& opt) {
+        opt == std::nullopt ? j = nullptr : j = *opt;
+    }
+
+    static void from_json(const json& j, std::optional<T>& opt) {
+        j.is_null() ? opt = std::nullopt : opt = j.get<T>();
+    }
+};
+
+template <typename T>
+struct adl_serializer<std::shared_ptr<T>> {
+    static void to_json(json& j, const std::shared_ptr<T>& ptr) {
+        !ptr ? j = nullptr : j = *ptr;
```

Comment:
I'd say `if`s are more readable, here

Reply:
Above is the worst way to do it, no question about that ;) We could `j = ptr ? *ptr : nullptr;` if we want ternary, but I think you're right and I'll go with `if`.